core/arch/arm/tee/tee_rpmb.c

```diff
@@ -212,31 +219,63 @@ static TEE_Result tee_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 	return TEE_SUCCESS;
 }
 
+#endif
+
 static TEE_Result tee_rpmb_key_gen(uint16_t dev_id __unused,
 				   uint8_t *key, uint32_t len, bool commercial)
 {
-	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_Result res = TEE_SUCCESS;
 	struct tee_hw_unique_key hwkey;
+	uint8_t *ctx = NULL;
 
-	if (key == NULL || RPMB_KEY_MAC_SIZE != len)
-		return TEE_ERROR_BAD_PARAMETERS;
+	if (key == NULL || RPMB_KEY_MAC_SIZE != len) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto out;
+	}
+
+#ifndef CFG_RPMB_TESTKEY
```

Comment:
I would prefer if we had two versions of the function instead with this #ifdef choosing the correct version.